### PR TITLE
[CI] Update latest ray version 2.5.0 -> 2.6.3

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
- Update all files to Ray version 2.6.3 as in https://github.com/ray-project/kuberay/pull/1151 
- Update the Ray version used in the sample Job yaml test to Ray 2.6.2 and remove old ray versions that we don't guarantee compatibility with.  This reduces costs and reduces the impact of flaky tests. We support the last three Ray stable versions and the last 1.x Ray version.

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .github/workflows/test-job.yaml [hunk 1]
```diff
@@ -307,53 +307,13 @@ jobs:
         with:
           ray_version: 1.13.0
 
-  test-compatibility-2_1_0:
```
Comment (architkulkarni) [inline .github/workflows/test-job.yaml:310]: The diff algorithm got confused, I just deleted 2.1-2.3 and added one for 2.6.2

**Review by kevin85421 — COMMENTED** [review]
Could you update all YAML files to 2.6.2? You can refer to #1151. Thanks!

**Comment by architkulkarni** [comment]
Sure, let me wait for 2.6.3 to come out (probably today)

**Comment by architkulkarni** [comment]
> Could you update all YAML files to 2.6.2? You can refer to #1151. Thanks!

@kevin85421 Done, please take a look

File: .github/workflows/test-job.yaml [hunk 2]
```diff
@@ -427,13 +387,13 @@ jobs:
         with:
           ray_version: nightly
 
-  sample-yaml-config-test-2_5_0:
+  sample-yaml-config-test-2_6_2:
```
Comment (kevin85421) [inline .github/workflows/test-job.yaml:390]: 2.6.3?

File: docs/guidance/batch-inference-example.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# RayJob Batch Inference Example
```
Comment (kevin85421) [inline docs/guidance/batch-inference-example.md:1]: What's this?

**Review by kevin85421 — APPROVED** [review]

File: docs/guidance/batch-inference-example.md (hunk 3 above)
Comment (architkulkarni) [reply docs/guidance/batch-inference-example.md:1]: Sorry I accidentally added file to git, let me remove it

File: .github/workflows/test-job.yaml (hunk 2 above)
Comment (architkulkarni) [reply .github/workflows/test-job.yaml:390]: Yup let me update it

**Comment by architkulkarni** [comment]
@kevin85421 it's expected that https://buildkite.com/ray-project/ray-ecosystem-ci-kuberay-ci/builds/1340#018a0039-6a6e-4187-bf7c-ed206f90b4bb/238-2911 will fail because it's a bug you fixed recently (after 0.6.0: https://github.com/ray-project/kuberay/pull/1332), right? In that case, I'll skip it for the latest-release test.

Actually, why do we need to test sample YAMLs from the tip of master against the latest release? Isn't it enough to test against the tip of the master branch?

**Comment by kevin85421** [comment]
> it's expected that https://buildkite.com/ray-project/ray-ecosystem-ci-kuberay-ci/builds/1340#018a0039-6a6e-4187-bf7c-ed206f90b4bb/238-2911 will fail because it's a bug you fixed recently (after 0.6.0: https://github.com/ray-project/kuberay/pull/1332), right? In that case, I'll skip it for the latest-release test.

Yes, sorry for the inconvenience. I forgot to rebase with the master branch before merging #1332.

> Actually, why do we need to test sample YAMLs from the tip of master against the latest release? Isn't it enough to test against the tip of the master branch?

We also need to test sample YAMLs with the latest release. Since most users utilize the latest release, it's crucial to ensure they don't encounter issues when running the sample YAMLs with it.

**Pull request merged**